nifi-nar-bundles/nifi-redis-bundle/nifi-redis-extensions/src/test/java/org/apache/nifi/redis/processor/TestPutRedisRecord.java

```diff
@@ -0,0 +1,221 @@
+import org.apache.nifi.util.TestRunners;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.stubbing.Answer;
+import org.springframework.data.redis.connection.RedisConnection;
+import org.springframework.data.redis.connection.RedisHashCommands;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class TestPutRedisRecord {
+
+    private TestRunner runner;
+    private PutRedisRecord processor;
+    private MockRecordParser parser;
+
+    private MockRedisConnectionPoolService connectionPoolService;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        processor = new PutRedisRecord();
+        runner = TestRunners.newTestRunner(processor);
+        parser = new MockRecordParser();
+        try {
+            runner.addControllerService("parser", parser);
+        } catch (InitializationException e) {
+            throw new IOException(e);
+        }
+        runner.enableControllerService(parser);
+        runner.setProperty(PutRedisRecord.RECORD_READER_FACTORY, "parser");
+
+        connectionPoolService = new MockRedisConnectionPoolService();
+        connectionPoolService.setFailAfterN(0);
+        try {
+            runner.addControllerService("connectionPool", connectionPoolService);
+        } catch (InitializationException e) {
+            throw new IOException(e);
+        }
+        runner.setProperty(connectionPoolService, RedisUtils.CONNECTION_STRING, "localhost:6379");
+        runner.setProperty(RedisUtils.REDIS_CONNECTION_POOL, "connectionPool");
+        // Tests should provide a field named 'hash' with unique values per record, unless testing failure conditions
+        runner.setProperty(PutRedisRecord.HASH_VALUE_RECORD_PATH, "/hash");
+    }
+
+    @Test
+    public void testPutRecords() {
+        runner.assertNotValid();
+        runner.enableControllerService(connectionPoolService);
+        parser.addSchemaField("hash", RecordFieldType.STRING);
+        parser.addSchemaField("name", RecordFieldType.STRING);
+        parser.addSchemaField("age", RecordFieldType.INT);
+
+        parser.addRecord("abc", "John Doe", 48);
+        parser.addRecord("def", "Jane Doe", 47);
+        parser.addRecord("ghi", "Jimmy Doe", 14);
+
+        runner.enqueue("hello".getBytes(StandardCharsets.UTF_8));
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred("success", 1);
+        final List<MockFlowFile> result = runner.getFlowFilesForRelationship("success");
+        assertEquals(1, result.size());
+        MockFlowFile ff = result.get(0);
+        ff.assertAttributeEquals("putredisrecord.successful.record.count", "3");
```

Comment:
Recommend replacing this attribute name with a new shared static value that can be reused.